lib/std/compress/zstandard.zig

```diff
@@ -20,20 +21,29 @@ pub fn DecompressStream(
     return struct {
         const Self = @This();
 
-        allocator: Allocator,
+        pub const window_size_max = options.window_size_max;
+
+        const table_size_max = types.compressed_block.table_size_max;
+
         source: std.io.CountingReader(ReaderType),
         state: enum { NewFrame, InFrame, LastBlock },
         decode_state: decompress.block.DecodeState,
         frame_context: decompress.FrameContext,
-        buffer: RingBuffer,
-        literal_fse_buffer: []types.compressed_block.Table.Fse,
-        match_fse_buffer: []types.compressed_block.Table.Fse,
-        offset_fse_buffer: []types.compressed_block.Table.Fse,
-        literals_buffer: []u8,
-        sequence_buffer: []u8,
+        buffer: WindowBuffer,
+        literal_fse_buffer: [table_size_max.literal]types.compressed_block.Table.Fse,
+        match_fse_buffer: [table_size_max.match]types.compressed_block.Table.Fse,
+        offset_fse_buffer: [table_size_max.offset]types.compressed_block.Table.Fse,
+        literals_buffer: [types.block_size_max]u8,
+        sequence_buffer: [types.block_size_max]u8,
```

Comment:
These being arrays rather than pointers makes the `DecompressStream` relatively large: 2^9 + 2^9 + 2^8 + 2^17 + 2^17 gives 263424 B = 257.25 KiB for these buffers.

Reply:
The size of the buffers can probably be cut in half by combining`literals_buffer` and `sequence_buffer`, though this would need double checking the spec to make sure it can be done and would add a bit of complexity to the compressed block handling.